ludwig/models/trainer.py

```diff
@@ -364,34 +368,152 @@ def train_for_tuning(
                     self.regularization_lambda
                 )
                 step_count += 1
-
-        # TODO (ASN) : return loss and learning rate for tune_learning_rate
         return model
 
-    def tune_batch_size(
+    def tune_learning_rate(
         self,
+        config,
         model,
         training_set,
+        random_seed: int = default_random_seed,
+        min_lr: float = 1e-8,
+        max_lr: float = 1.0,
+        total_training_steps: int = 100,
```

Comment:
this seems like a big number of steps

Reply:
@w4nderlust totally agree. I pulled this default value from the pytorch lightning LR tuner. We can modify our implementation is you think something smaller is more appropriate.

Reply:
We are spanning through 8 orders of magnitude with the default parameters, probably less than 10 for each order of magnitude may be sufficient. Although let's keep it like this for now and if we figure out that it's too slow, 50 would probably be sufficient too.